File: Igtampe.CDBFS.Common/CdbfsDirectory.cs
namespace Igtampe.CDBFS.Common {
    public class CdbfsDirectory {

        public CdbfsDrive? Drive { get; set; }
        public CdbfsFolder? Folder { get; set; }
        
        public List<CdbfsFolder> Subfolders { get; set; } = [];
        public List<CdbfsFile> Files { get; set; } = [];

        public int FileCount => Files.Count;
        public int FolderCount => Subfolders.Count;
        public long FileSize => Files.Sum(a => a.Size);
        public long FoldersSize => Subfolders.Sum(a => a.Size);
        public long FolderSize => Folder?.Size ?? 0;
        public long DriveSize => Drive?.Size ?? 0;


    }
}

File: igtampe.cdbfs.frontend/src/model/CdbfsDirectory.ts
import CdbfsDrive from "./CdbfsDrive";
import CdbfsFile from "./CdbfsFile";
import CdbfsFolder from "./CdbfsFolder";

export default class CdbfsDirectory{
    public constructor(
        public drive : CdbfsDrive,
        public folder : CdbfsFolder,
        public subfolders : CdbfsFolder[],
        public files : CdbfsFile[],
        public fileCount:number,
        public folderCount:number,
        public fileSize:number,
        public foldersSize: number,
        public folderSize:number,
        public driveSize:number
    ){}
}